From b564776250eaff47728eb63c25bbefcfd1c0c240 Mon Sep 17 00:00:00 2001
From: Jeff Moyer <jmoyer@redhat.com>
Date: Thu, 31 Jan 2013 15:57:02 -0500
Subject: [PATCH 2/2] fix test case 16 to work on arm64

aarch64 does not implement "legacy" system calls such as eventfd.  It
does, of course, support eventfd2, so use that instead.

Reported-by: Riku Voipio <riku.voipio@linaro.org>
Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
---
 harness/cases/16.t |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/harness/cases/16.t b/harness/cases/16.t
index c3157cc..5a546ff 100644
--- a/harness/cases/16.t
+++ b/harness/cases/16.t
@@ -18,6 +18,12 @@
 #define SYS_eventfd 318
 #elif defined(__alpha__)
 #define SYS_eventfd 478
+#elif defined(__aarch64__)
+/* arm64 does not implement eventfd, only eventfd2 */
+#define USE_EVENTFD2
+#ifndef SYS_eventfd2
+#define SYS_eventfd2 19
+#endif /* __aarch64__ */
 #else
 #error define SYS_eventfd for your arch!
 #endif
@@ -39,7 +45,11 @@ int test_main(void)
 	struct timespec	notime = { .tv_sec = 0, .tv_nsec = 0 };
 
 	buf = malloc(SIZE);				assert(buf);
+#ifndef USE_EVENTFD2
 	efd = syscall(SYS_eventfd, 0);
+#else
+	efd = syscall(SYS_eventfd2, 0, 0);
+#endif
 	if (efd < 0) {
 		if (errno == ENOSYS) {
 			printf("No eventfd support.  [SKIPPING]\n");
-- 
1.7.10.4

